pkg/uprobetracer/tracer.go

```diff
@@ -218,7 +216,7 @@ func (t *Tracer[Event]) attach(containerPid uint32) {
 	}
 
 	for _, filePath := range attachFilePaths {
-		file, err := os.OpenFile(filePath, unix.O_PATH, 0)
+		file, err := os.Open(filePath)
```

Comment:
I remember you removed the `O_PATH` because `FOp` was `empty_fops` in that case:
https://github.com/torvalds/linux/blob/v6.2/fs/open.c#L839-L841

But `pkg/uprobetracer` does not use `FOp` but only `RealInode`. Does it work if you keep `O_PATH`?

Reply:
If I use `O_PATH`, the `private_data` will also be `0`, then I'm not able to access the real file.

Reply:
Ah, right. Could you add a comment above explaining why O_PATH cannot be used?